File: user_url.py
from flask import Blueprint, request, Response
from .activity_facade import ActivityFacade

user_bp = Blueprint('user', __name__)
facade = ActivityFacade()

@user_bp.route('/deploy-atividade', methods=['GET'])
def deploy_activity():
    invenira_instance_id = request.args.get('invenira_instance_id')
    activity_type = request.args.get('activity_type')

    if not all([invenira_instance_id, activity_type]):
        return Response("Erro: parâmetros em falta.", status=400)

    try:
        deployment_url = facade.get_deployment_url(
            activity_type,
            invenira_instance_id,
            request.host_url
        )
        return Response(deployment_url, mimetype='text/plain')
    except ValueError as e:
        return Response(str(e), status=404)

File: user_url.php
<?php
// user_url: http://<domínio>/deploy-atividade?invenira_instance_id=...
header('Content-Type: text/plain');
$invenira_instance_id = $_GET['invenira_instance_id'] ?? null;

if ($invenira_instance_id) {
    // Ação interna: Preparar-se para guardar analytics para esta instância da atividade.
    // Exemplo: Criar uma entrada na base de dados do AP para rastrear esta instância.
    // ...
    
    // Devolver o URL a ser usado pelos estudantes.
    $deployment_url = "http://<dominio_ap>/atividade_instancia/{$invenira_instance_id}";
    echo $deployment_url;
} else {
    http_response_code(400); // Bad Request
    echo "Erro: ID da instância da Inven!RA em falta.";
}
?>